### PR TITLE
Fix BlackholePageSink

### PR DESCRIPTION
**Pull request by arhimondr**
Presto may call `PageSink#finish()` before the latest write is finished.
Connector should take care of such situation itself, returning the same
blocked future that write is waiting on.

### CODE REVIEW

**Comment by martint** [comment]
> Presto may call PageSink#finish() before the latest write is finished.

That seems like a bug, actually.

File: presto-spi/src/main/java/com/facebook/presto/spi/ConnectorPageSink.java [hunk 1]
```diff
@@ -29,6 +29,10 @@
      */
     CompletableFuture<?> appendPage(Page page);
 
+    /**
```
Comment (electrum) [inline presto-spi/src/main/java/com/facebook/presto/spi/ConnectorPageSink.java]: How about this
```java
    /**
     * Notifies the connector that no more pages will be appended and returns
     * connector-specific information that will be sent to the coordinator to
     * complete the write operation. This method may be called immediately
     * after the previous call to {@link #appendPage} (even if the returned
     * future is not complete).
     */

**Comment by arhimondr** [comment]
@martint 

> That seems like a bug, actually.

This is the way how Presto works. `finish()` only supposed to notify the connector that there will be no more append calls. But shouldn't ensure that all the appends are actually finished. We were discussing this with @electrum . Could you please discuss it with him?

File: presto-spi/src/main/java/com/facebook/presto/spi/ConnectorPageSink.java (hunk 1 above)
Comment (arhimondr) [inline presto-spi/src/main/java/com/facebook/presto/spi/ConnectorPageSink.java]: That sounds much better. Thanks!

**Comment by electrum** [comment]
The question is semantics of the returned future. Does it mean append is finished, or whether future append calls can be made (is blocked)? The docs say the latter, which matches the implementation, so I think it's best just to clarify the docs.

**Comment by electrum** [comment]
Merged, thanks!

**Pull request closed without merging**